confusing error when lcms2 is missing

### What is the bug?

Unfortunately it doesn't appear that find_library() supports the REQUIRED keyword. As when lcms2 is missing it doesn't give any useful error message. And instead wrongly complains about PRIVATE.

```
CMake Error at cmake/helpers/GdalDriverHelper.cmake:390 (message):
  gdal_target_link_libraries(): PRIVATE is a mandatory argument.
Call Stack (most recent call first):
  frmts/pdf/CMakeLists.txt:79 (gdal_target_link_libraries)
```

https://github.com/OSGeo/gdal/blob/3aa4582eb482f77937957428fe71e934826b9c02/frmts/pdf/CMakeLists.txt#L82

### Steps to reproduce the issue

`cmake .. -DGDAL_USE_PDFIUM=ON`

### Versions and provenance

all

### Additional context

_No response_

--- a/frmts/pdf/CMakeLists.txt
+++ b/frmts/pdf/CMakeLists.txt
@@ -79,7 +79,10 @@ if (GDAL_USE_PDFIUM)
     find_package(JPEG REQUIRED)
     find_package(PNG REQUIRED)
     find_package(OpenJPEG REQUIRED)
-    find_library(LCMS2_LIBRARY NAMES lcms2 REQUIRED)
+    find_library(LCMS2_LIBRARY NAMES lcms2)
+    if(NOT LCMS2_LIBRARY)
+      message(FATAL_ERROR "LCMS2 library not found. Please install liblcms2-dev.")
+    endif()
 
     # Rather hacky... Related how we build pdfium in https://github.com/rouault/pdfium_build_gdal_3_4
     gdal_target_link_libraries(

